### PR TITLE
remove-support-for-dns-mode-private

### PR DESCRIPTION
**Pull request by calvix**
towartds https://github.com/giantswarm/roadmap/issues/2690

### CODE REVIEW

**Review by fiunchinho — CHANGES_REQUESTED** [review]
I think we can remove this now https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/templates/_aws_cluster.tpl#L12-L16

And also the `additionalVpc` field from the schema and values files.

**Comment by calvix** [comment]
> I think we can remove this now [`master`/helm/cluster-aws/templates/_aws_cluster.tpl#L12-L16](https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/templates/_aws_cluster.tpl?rgh-link-date=2023-09-20T15%3A05%3A13Z#L12-L16)
> 
> And also the `additionalVpc` field from the schema and values files.

removed

File: helm/cluster-aws/values.schema.json [hunk 1]
```diff
@@ -225,26 +225,12 @@
                     "type": "object",
                     "title": "DNS",
                     "properties": {
-                        "additionalVpc": {
```
Comment (AndiDog) [inline helm/cluster-aws/values.schema.json:228]: You need to run

```
make normalize-schema generate-values generate-docs validate-schema
```

**Review by AndiDog — CHANGES_REQUESTED** [review]

File: helm/cluster-aws/values.schema.json (hunk 1 above)
Comment (calvix) [reply helm/cluster-aws/values.schema.json:228]: I always run that, otherwise the  `JSON schema validation` CI would fail

**Review by fiunchinho — COMMENTED** [review]
Actually, would it make sense to remove the whole `dns` block? I don't think we need it anymore, right? https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/values.schema.json#L224-L265

**Comment by calvix** [comment]
> Actually, would it make sense to remove the whole `dns` block? I don't think we need it anymore, right? [`master`/helm/cluster-aws/values.schema.json#L224-L265](https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/values.schema.json?rgh-link-date=2023-09-21T08%3A06%3A23Z#L224-L265)

Don't we still use the resolver rules in the product?

File: helm/cluster-aws/values.schema.json (hunk 1 above)
Comment (calvix) [reply helm/cluster-aws/values.schema.json:228]: Oh i was missing `generate-docs` I wonder why that command is missing from the list in the Readme, just added it

Comment (calvix) [reply helm/cluster-aws/values.schema.json:228]: sorry my bad

**Comment by fiunchinho** [comment]
> > Actually, would it make sense to remove the whole `dns` block? I don't think we need it anymore, right? [`master`/helm/cluster-aws/values.schema.json#L224-L265](https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/values.schema.json?rgh-link-date=2023-09-21T08%3A06%3A23Z#L224-L265)
> 
> Don't we still use the resolver rules in the product?

I think we only needed it when using private DNS, but if we are dropping support for private dns, I don't think we need it anymore. I may be wrong though.

**Comment by calvix** [comment]
@T-Kukawka DO we wanna keep a feature called Resolver rules in our product or do we wanna get rid of it too? 
https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resolver.html

**Review by AndiDog — APPROVED** [review]

**Comment by T-Kukawka** [comment]
Personally i would leave this feature if it does not bring much of overhead on our side, especially for the future to target more 'hybrid' customers where this can come in handy.

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**